myfitnesspal/client.py

```diff
@@ -996,6 +990,7 @@ def set_new_goal(
 
         # Get User Default Unit Preference
         if energy_unit != "calories" and energy_unit != "kilojoules":
+            assert self.user_metadata
```

Comment:
Just to keep the linter happy; `self.user_metadata` is technically marked `Optional`, but you're right that you can reasonably rely on it being set.